rclcpp/src/rclcpp/executor.cpp

```diff
@@ -38,14 +41,23 @@
 
 using namespace std::chrono_literals;
 
-using rclcpp::exceptions::throw_from_rcl_error;
 using rclcpp::Executor;
 
+/// Mask to indicate to the waitset to only add the subscription.
+/// The events and intraprocess waitable are already added via the callback group.
+static constexpr rclcpp::SubscriptionWaitSetMask kDefaultSubscriptionMask = {true, false, false};
+
 Executor::Executor(const rclcpp::ExecutorOptions & options)
 : spinning(false),
   interrupt_guard_condition_(std::make_shared<rclcpp::GuardCondition>(options.context)),
   shutdown_guard_condition_(std::make_shared<rclcpp::GuardCondition>(options.context)),
-  memory_strategy_(options.memory_strategy)
+  notify_waitable_(std::make_shared<rclcpp::executors::ExecutorNotifyWaitable>(
```

Comment:
I see this is being used with the `collector_` and the `current_notify_waitable_`, but I always like to ask if we really need this to be a shared pointer? Is it possible to avoid that? If not that's ok. I don't have enough context to say easily.

Reply:
I suppose it's nicer than having a non-owning pointer for `current_notify_waitable_` and passing a reference to `collector_`?

Reply:
I think at some point that the `ExecutorNotifyWaitable` could also be derived from for other executor implementations, so keeping a pointer to the base was the idea here.  It may be that we don't need that anymore as this implementation does everything we need for the waitset and events executors (cc @alsora)